### ISSUE TITLE
New checks to ensure all OS/2 metadata is consistent and complete

### ISSUE THREAD

**Issue by davelab6** [fonttools/fontbakery#1619]
@getflourish published the amazing https://medium.com/@getflourish/the-anatomy-of-a-thousand-typefaces-f7b9088eed1 and says, emphasis mine,

> While some fonts provide a lot of information, this isn’t always the case, **especially not for free or open source fonts such as the ones from Google Fonts.** But even if information is provided, it’s not necessarily consistent or complete enough to make comparisons based on it.

We should fix that :) 

I took a look at the dictionary data powering his impressive https://getflourish.github.io/anatomy-of-typefaces/app/ and he includes the values of the following OS/2 Table keys:

- [ ] xAvgCharWidth
- [ ] usWidthClass
- [ ] ySubscriptXSize
- [ ] ySubscriptYSize
- [ ] ySubscriptXOffset
- [ ] ySubscriptYOffset
- [ ] ySuperscriptXSize
- [ ] ySuperscriptYSize
- [ ] ySuperscriptXOffset
- [ ] ySuperscriptYOffset
- [ ] yStrikeoutSize
- [ ] yStrikeoutPosition
- [ ] sFamilyClass
- [ ] panose
- [ ] ulCodePageRange1
- [ ] ulCodePageRange2
- [ ] sxHeight
- [ ] sCapHeight
- [ ] usDefaultChar

We should have checks that ensure all of these are consistent and complete across our whole collection.

**Comment by madig** [comment]
How are you going to check Panose? He doesn't seem to compute that.

**Comment by getflourish** [comment]
Right, one of my main points about missing information is the lack of Panose. Since most of its values need to be manually calculated and provided by the designer, I don’t think that this is easily possible to add. Technically, all measurements could be done algorithmically though.

**Comment by madig** [comment]
You could, of course, work out a CAPTCHA thing that makes people classify things, e.g. by showing them comparison images where you know the classification 😁

**Comment by felipesanches** [comment]
I still need to look more carefully at this one, but this is definitely not worth blocking the v0.5.1 - MFDOS release.
